avocado/core/data_dir.py

```diff
@@ -193,15 +193,23 @@ class _TmpDirTracker(Borg):
 
     def __init__(self):
         Borg.__init__(self)
+        self.keep_tmp = False
+
+    def get(self, basedir, keep_tmp):
+        # If someone sets the tmp_dir to be kept, further
+        # calls to get()  will not unset that.
+        if not self.keep_tmp:
+            self.keep_tmp = keep_tmp
 
-    def get(self):
         if not hasattr(self, 'tmp_dir'):
-            self.tmp_dir = tempfile.mkdtemp(prefix='avocado_')
+            self.tmp_dir = tempfile.mkdtemp(prefix='avocado_',
+                                            dir=basedir)
```

Comment:
Actually I think users can use basedir withouth setting keep_tmp  to True. That's a valid use case, otherwise we don't need the `keep_tmp` at all.